abci.go

```diff
@@ -111,9 +111,12 @@ func (c CheckResult) ToABCI() abci.ResponseCheckTx {
 	}
 }
 
-// TickResult allows the Ticker to modify the validator set
+// TickResult contains the result of the tasks execution.
 type TickResult struct {
-	Diff []ValidatorUpdate
+	// List of task IDs that were executed. Task is considered executed
+	// when processing it caused any change to the state (even if it is
+	// only removing the task from the queue and no other change).
+	Executed [][]byte
```

Comment:
1. I have removed the validator update diff because it is not needed right now. I was also wondering if it makes sense to have it here at all, as you can schedule execution of the [validator update message](https://github.com/iov-one/weave/blob/master/x/validators/codec.proto#L9-L12) and achieve the same using the generic solution. Should I add this attribute back anyway?

2. I am not sure if I understand what you mean correctly. Are you saying that requiring every result to return an ID is specific to the `x/cron` implementation? If so, then I think you are right :sweat_smile: I wonder if it makes sense to have a different cron implementation anyway.
This attribute can be empty as it is not required. Other implementations could ignore this field.

I am not sure if I understand the last point as well. Should I change `TickResult` to contain only tags and validator update diff attributes?

Reply:
Implemented in https://github.com/iov-one/weave/pull/850/commits/8b48997ed0e97d8c8e834af083b400524aa01bed

Please check if the interface is good enough. I am returning tags and validators diff separately,  instead of wrapping them in a structure.

Reply:
The `type TickResult struct` is not used anymore. Better remove it.

Reply:
Done in https://github.com/iov-one/weave/pull/850/commits/0cb9044a5c2b4006344a4c44aef3052cc7572488